File: contest/a_boats_and_ferries/sol/a_boats_and_ferries.py
# read and store p, b and f as integers from the input (3 lines)
p = int(input())
b = int(input())
f = int(input())

# print the expected result, that is to say:
# p - (500 x f + 4 x b) if this number is positive;
# 0 otherwise
print(max(p - (500*f + 4*b), 0))

File: contest/a_boats_and_ferries/sol/a_boats_and_ferries.java
import java.util.Scanner;

public class cb1 {

    public static void main(String[] args) {

        // read and store the input numbers as integers
        Scanner sc = new Scanner(System.in);
        int p = sc.nextInt();
        int b = sc.nextInt();
        int f = sc.nextInt();

        // compute the number of peple who won't fit in the boats or ferries
        int r = p - (4 * b + 500 * f);

        // if this number is negative, make it 0 instead
        if (r < 0){
            r = 0;
        }

        // then print the result
        System.out.println(r);

    }

}
